ent/encryption/utils.cc

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2025 ScyllaDB
+ *
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include <seastar/net/tls.hh>
+#include <seastar/net/dns.hh>
+#include <seastar/util/short_streams.hh>
+
+#include "utils.hh"
+
+using namespace seastar;
+
+encryption::httpclient::httpclient(std::string host, uint16_t port, seastar::shared_ptr<seastar::tls::certificate_credentials> creds)
+    : _host(std::move(host))
+    , _port(port)
+    , _creds(std::move(creds))
+    , _req(http::request::make(httpd::operation_type::GET, _host, ""))
+{
+    _req._version = "1.1";
+}
+
+encryption::httpclient& encryption::httpclient::add_header(std::string_view key, std::string_view value) {
+    _req._headers[sstring(key)] = sstring(value);
+    return *this;
+}
+
+void encryption::httpclient::clear_headers() {
+    _req._headers.clear();
+}
+
+seastar::future<encryption::httpclient::result_type> encryption::httpclient::send() {
+    result_type res;
+    co_await send([&](const http::reply& r, std::string_view body) {
+        res.reply._status = r._status;
+        res.reply._content = sstring(body);
+        res.reply._headers = r._headers;
+        res.reply._version = r._version;
+    });
+    co_return res;
+}
+
+/**
+ * Note: the client object is a one-off send-recive. We don't keepalive across calls or anything.
+ * This because 
```

Comment:
might want to put

> This is because

Reply:
No, that changes the inflection of the comment.

Reply:
:-)